src/model/core/fuel.jl

```diff
@@ -0,0 +1,117 @@
+
+From above formulations, thermal generators are expected to have the same fuel consumption per generating 1 MWh electricity, regardless of minimum load or full load. However, thermal generators tend to have decreased efficiency when operating at part load, leading to higher fuel consumption per generating the same amount of electricity. To have more precise representation of fuel consumption at part load, the piecewise-linear fitting of heat input can be introduced. 
+
+```math
+\begin{aligned}
+vFuel_{y,t} >= vP_{y,t} * h_{y,x} + U_{g,t}* f_{y,x}
+\hspace{1cm} \forall y \in G, \forall t \in T, \forall x \in X
+\end{aligned}
+```
+Where $h_{y,x}$ represents incremental heat rate of a thermal generator $y$ in segment $x$ [MMBTU/MWh] and $f_{y,x}$ represents intercept of fuel consumption of a thermal generator $y$ in segment $x$ [MMBUT], and $U_{y,t}$ represents the commit status of a thermal generator $y$ at time $t$. We include at most three segments to represent the piecewise heat consumption. 
+
+Since fuel consumption has a positive value, the optimization will optimize the fuel consumption by enforcing the inequity to equal to the highest piecewise segment. When the power output is zero, the commitment variable $U_{g,t}$ will bring the intercept to be zero such that the fuel consumption is zero when thermal units are offline.
+
+In order to run piecewise fuel consumption module, the unit commitment must be turned on, and users should provide Incremental_Heat_Rate_Segmenti and Intercept_Fuel_Consumption_Segmenti for at least one segment. 
+
+If users only want to model piecewise heat rate for some of thermal generators, then set the Incremental_Heat_Rate_Segment1 for thoese plants to be the same as conventional heat rate, and then set Incremental_Heat_Rate_Segment2/3 and Intercept_Fuel_Consumption_Segment1/2/3 to be zero.
+"""
+
+function fuel!(EP::Model, inputs::Dict, setup::Dict)
+    println("Fuel Module")
+    dfGen = inputs["dfGen"]
+    T = inputs["T"]     # Number of time steps (hours)
+    Z = inputs["Z"]     # Number of zones
+    G = inputs["G"]
+    THERM_COMMIT = inputs["THERM_COMMIT"]
+    FUEL = length(inputs["fuels"])
+    ALLGEN = collect(1:G)
+    # create variable for fuel consumption for output
+    @variable(EP, vFuel[y in 1:G, t = 1:T] >= 0)
+    
+    ### Expressions ####
+    # Fuel consumed on start-up (MMBTU or kMMBTU (scaled)) 
+    # if unit commitment is modelled
+    @expression(EP, eStartFuel[y in 1:G, t = 1:T],
+        if y in THERM_COMMIT
+            (dfGen[y,:Cap_Size] * EP[:vSTART][y, t] * 
+                dfGen[y,:Start_Fuel_MMBTU_per_MW])
+        else
+            EP[:vZERO]
+        end)
+
+    # fuel_cost is in $/MMBTU (M$/billion BTU if scaled)
+    # vFuel and eStartFuel is MMBTU (or billion BTU if scaled)
+    # Therefore eCFuel_start or eCFuel_out is $ or Million$)
+    # Separately track the start up fuel and fuel consumption for power generation
```

Comment:
What is this sentence referring to?

Reply:
just rambling about the units (w/ or w/o scaling)..